### PR TITLE
[7.7.x] RHPAM-1281: Stunner - Support for image strips

### PR DESCRIPTION
**Pull request by romartin**
Hey @hasys 

See https://issues.jboss.org/browse/RHPAM-1281

This is an important performance improvement for next 7.0.2, please can you give it priority?

It depends on:
* https://github.com/kiegroup/lienzo-core/pull/48
* https://github.com/kiegroup/lienzo-tests/pull/31

Thanks!!

### CODE REVIEW

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
@hasys the amount of files changed in this PR is due to I updated all the icons for bpmn and stunner common images and also removed some unused code (palette). Sorry..

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/test/java/org/kie/workbench/common/stunner/client/lienzo/components/glyph/LienzoImageStripLoaderTest.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        Command callback = mock(Command.class);
+        tested.init(metadata,
+                    callback);
+        ArgumentCaptor<com.ait.lienzo.client.core.image.ImageStrip[]> stripsCaptor =
+                ArgumentCaptor.forClass(com.ait.lienzo.client.core.image.ImageStrip[].class);
+        verify(lienzoStripRegistration, times(1)).accept(stripsCaptor.capture(),
+                                                         any(Runnable.class));
+        final com.ait.lienzo.client.core.image.ImageStrip[] strips = stripsCaptor.getValue();
+        assertNotNull(strips);
+        assertTrue(strips.length == 1);
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/test/java/org/kie/workbench/common/stunner/client/lienzo/components/glyph/LienzoImageStripLoaderTest.java]: `assertEquals(1, strips.length);` can be used instead.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/resources/BPMNImageResources.java [hunk 2]
```diff
@@ -71,6 +71,15 @@
...
     DataResource categoryServiceTasks();
 
+    // **** Icons Sprite ****
+    @ClientBundle.Source("images/icons/bpmn-icons-sprite.png")
+    ImageResource bpmnIconsSprite();
+
+    @ClientBundle.Source("images/icons/bpmn-icons-sprite.css")
+    BPMNCssResource bpmnIconsSpriteCss();
+
+    // TODO: CLEAR UNUSED RESOURCES
```
[the remark this answers is not in the document]
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/java/org/kie/workbench/common/stunner/bpmn/client/resources/BPMNImageResources.java]: Well no even need for a new Jira ticket.. I just did the clean, this way all those static resources are not included in the webapp, anyway they're not used right now, as it's using the image from the "sprite", so good catch :+1: 
Thx!

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-client/src/main/java/org/kie/workbench/common/stunner/cm/client/shape/def/CaseManagementTaskShapeDef.java [hunk 3]
```diff
@@ -17,17 +17,21 @@
...
 import org.kie.workbench.common.stunner.cm.client.shape.view.ActivityView;
+import org.kie.workbench.common.stunner.core.client.shape.ImageDataUriGlyph;
 import org.kie.workbench.common.stunner.core.definition.shape.ShapeDef;
 
 public final class CaseManagementTaskShapeDef
         implements CaseManagementActivityShapeDef<BaseTask, ActivityView> {
 
+    ImageDataUriGlyph NONE_TASK_GLYPH = ImageDataUriGlyph.create(BPMNImageResources.INSTANCE.task().getSafeUri());
+
+    // TODO: Use the rigth icon for the task type
```
[the remark this answers is not in the document]
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-case-mgmt/kie-wb-common-stunner-case-mgmt-client/src/main/java/org/kie/workbench/common/stunner/cm/client/shape/def/CaseManagementTaskShapeDef.java]: Fixed

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/preferences/StunnerPreferencesRegistryLoader.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    public void load(final String definitionSetId,
+                     final ParameterizedCommand<StunnerPreferences> loadCompleteCallback,
+                     final ParameterizedCommand<Throwable> errorCallback) {
+        final Annotation qualifier = definitionUtils.getQualifier(definitionSetId);
+        load(qualifier,
+             loadCompleteCallback,
+             errorCallback);
+    }
+
+    public void load(final Annotation qualifier,
```
[the remark this answers is not in the document]
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/preferences/StunnerPreferencesRegistryLoader.java]: Fixed

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/shape/ImageStripRegistry.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+    public ImageStrip[] get(final Annotation... qualifiers) {
+        final List<ImageStrip> result = new LinkedList<>();
+        for (Annotation qualifier : qualifiers) {
+            final ManagedInstance<ImageStrip> i = stripInstances.select(qualifier);
+            if (!i.isUnsatisfied() && !i.isAmbiguous()) {
+                result.add(i.get());
+            }
+        }
+        return result.stream().toArray(ImageStrip[]::new);
```
[the remark this answers is not in the document]
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/shape/ImageStripRegistry.java]: Fixed

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/test/java/org/kie/workbench/common/stunner/client/lienzo/components/glyph/LienzoImageStripLoaderTest.java (hunk 1 above)
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/test/java/org/kie/workbench/common/stunner/client/lienzo/components/glyph/LienzoImageStripLoaderTest.java]: Fixed

**Comment by romartin** [comment]
Hey @hasys 

Thanks for the rewview, good catches!!  :+1: 

I guess I fixed all your comments and the icons are not being "blurred" now, so please give it a try and lemme know!

Answering your Q about the icons not used now - I just prefer to keep the source icons (image files) in the repository, because if we have to update or regenerate the sprite at some point, we don't have to scale each icon again from the svg source file, so there exist a png as well which is also scaled to the right side and quality. On the other hand, it's true it was missing to remove the unused `ImageResource`'s from the client code for all these unused icons, this way it has no impact on the generated artifacts, as the icons are not being included in the build, so fixed.

**Comment by romartin** [comment]
Hey @hasys 
Also fixed and added tests on the lienzo side, so I guess it's all done now, please lemme know and thanks!! :+1:

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
hey @hasys 
mmmm good catch, I did not realized about the icons during drag from the palette... good point! They're burred because they're being scaled (bigger), so I guess I can fix it jus tby changing the size, but anyway I'm agree with you as well, let's better add this improvement and lemme report it for further fixing.

**Comment by romartin** [comment]
Okk so created [https://issues.jboss.org/browse/JBPM-7400](https://issues.jboss.org/browse/JBPM-7400), thanks @hasys , good catch!

**Pull request merged**